### ISSUE TITLE
Bad decoding cause by splitted multi-bytes characters

### ISSUE THREAD

**Issue by damascene** [ProgVal/Limnoria#1232]
Limnoria does not fetch non-Latin text correctly. it only fetch part of the text and does not show the (1 more messages) as it does when the text is only Latin characters.
to test this with a Russian text:

```
@fetch https://paste.progval.net/raw/1689/
```

It was also tested with Arabic and Hebrew

**Comment by damascene** [comment]
Some comments from @ProgVal 

> you see, in most common encodings of Unicode (UTF-8 and UTF-16), characters can take from 1 to 4 bytes
> so in the same text, characters don't have the same size and `@fetch` gets only the X first *bytes* and sometimes, it splits a character code in half. 

here are characters that fit on one byte:
```
!"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_`abcdefghijklmnopqrstuvwxyz{|}~
```

**Comment by GLolol** [comment]
I'm not too familiar with how Supybot handles encoding, but would it be better to cut text by characters instead of bytes?

**Comment by ProgVal** [comment]
It is not related to Supybot.
The problem is that a character is truncated while downloading the page. But we cannot know the encoding of a page before it is downloaded.

**Comment by ProgVal** [comment]
This issue was caused by string length being calculated in characters instead of bytes, while IRC servers truncate on bytes. Fixed some time ago by 63a17f7491dbef5268e7bb3715bf12d9137a0ace
